src/SignalR/server/Core/src/HubInvocationContext.cs

```diff
@@ -24,14 +25,17 @@ public class HubInvocationContext
         /// <param name="hub">The instance of the Hub.</param>
         /// <param name="hubMethod">The <see cref="MethodInfo"/> for the Hub method being invoked.</param>
         /// <param name="hubMethodArguments">The arguments provided by the client.</param>
-        public HubInvocationContext(HubCallerContext context, IServiceProvider serviceProvider, Hub hub, MethodInfo hubMethod, object[] hubMethodArguments)
-#pragma warning disable CS0618 // Type or member is obsolete
-            : this(context, hubMethod.Name, hubMethodArguments)
-#pragma warning restore CS0618 // Type or member is obsolete
+        public HubInvocationContext(HubCallerContext context, IServiceProvider serviceProvider, Hub hub, MethodInfo hubMethod, IReadOnlyList<object> hubMethodArguments)
         {
             Hub = hub;
             ServiceProvider = serviceProvider;
             HubMethod = hubMethod;
+            Arguments = hubMethodArguments as object[] ?? hubMethodArguments.ToArray();
```

Comment:
We should be doing this when invoking the method, not creating the context. We should just store the object passed in as is without a copy.